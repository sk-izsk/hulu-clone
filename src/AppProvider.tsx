import React from 'react';
import { ThemeProvider } from 'react-jss';
import { theme } from './theme';

export interface AppProviderProps {}

const AppProvider: React.FC<AppProviderProps> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export { AppProvider };
