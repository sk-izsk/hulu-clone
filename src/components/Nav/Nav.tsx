import React from 'react';
import { createUseStyles } from 'react-jss';
import { CustomTheme } from '../../theme/theme';
import { navBarData } from '../../utils';

export interface NavProps {}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  nav: {
    display: 'flex',
    color: theme.colors.secondaryColor,
    overflowX: 'auto',
    padding: theme.spacing(1.25),
    paddingLeft: theme.spacing(5),
  },
  navItem: {
    cursor: 'pointer',
    marginLeft: theme.spacing(3.75),
    marginRight: theme.spacing(3.75),
    fontWeight: 500,
    opacity: 0.7,
    transition: 'transform 50ms ease-out',
    '&:hover': {
      opacity: 1,
      transform: 'scale(1.2)',
    },
  },
}));

const Nav: React.FC<NavProps> = () => {
  const classes = useStyles();
  return (
    <div className={classes.nav}>
      {navBarData.map((navData: string) => (
        <h2 className={classes.navItem} key={navData}>
          {navData}
        </h2>
      ))}
    </div>
  );
};
export { Nav };
