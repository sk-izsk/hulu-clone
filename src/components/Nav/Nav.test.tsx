import { render } from '@testing-library/react';
import React from 'react';
import { AppProvider } from '../../AppProvider';
import { Nav } from './Nav';

test('Nav renders correctly', () => {
  const { asFragment } = render(
    <AppProvider>
      <Nav />
    </AppProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
