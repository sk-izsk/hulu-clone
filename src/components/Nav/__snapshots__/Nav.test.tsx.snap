// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<DocumentFragment>
  <div
    class="nav-0-2-1"
  >
    <h2
      class="navItem-0-2-2"
    >
      Trending
    </h2>
    <h2
      class="navItem-0-2-2"
    >
      Top Rated
    </h2>
    <h2
      class="navItem-0-2-2"
    >
      Action
    </h2>
    <h2
      class="navItem-0-2-2"
    >
      Comedy
    </h2>
    <h2
      class="navItem-0-2-2"
    >
      Horror
    </h2>
    <h2
      class="navItem-0-2-2"
    >
      Romance
    </h2>
    <h2
      class="navItem-0-2-2"
    >
      Mystery
    </h2>
    <h2
      class="navItem-0-2-2"
    >
      Sci-fi
    </h2>
    <h2
      class="navItem-0-2-2"
    >
      Western
    </h2>
    <h2
      class="navItem-0-2-2"
    >
      Animation
    </h2>
  </div>
</DocumentFragment>
`;
